src/core/server/http/router/router.ts

```diff
@@ -341,10 +349,11 @@ export type RequestHandler<
   P = unknown,
   Q = unknown,
   B = unknown,
+  Context extends RequestHandlerContext = RequestHandlerContext,
```

Comment:
Order matters. Lots of plugins provide arguments to `RequestHandler` generic manually.

Reply:
> Lots of plugins provide arguments to `RequestHandler` generic manually.

This is a bit surprising to me. I'm curious why this is happening / needed vs. using the inferred typings from the validation schema.

Reply:
there are several cases:
- a plugin [implements a wrapper](https://github.com/elastic/kibana/blob/b3b0f1b0a8acd4bdadb65f88f39e63e31d6775eb/x-pack/plugins/beats_management/server/routes/wrap_route_with_security.ts#L11) around request handler and declare its type manually instead if using `RequestHandlerWrapper` type provided by the Core
- a plugin [decouple](https://github.com/elastic/kibana/blob/3b728b73cf5720b48e9759c7bb8b845e18ba5b57/x-pack/plugins/fleet/server/routes/agent_policy/handlers.ts#L37) request handler declaration from a route declaration
- a plugin [implements a custom route registration mechanism](https://github.com/elastic/kibana/blob/adab1bc991ba12a5d5e91cba0c3f5312a036284f/x-pack/plugins/infra/server/lib/adapters/framework/kibana_framework_adapter.ts#L43)